### PR TITLE
(core) sort keys when diffing json

### PR DESCRIPTION
**Pull request by anotherchrisberry**
I found myself looking at the revision history of some pipelines yesterday and it was pretty difficult because, when we save a pipeline, Front50 does not use sorted Maps, so properties may end up on different lines between saves. Since Deck is doing a dumb comparison, it's hard to tell which properties have actually changed between versions.

We could fix the problem in Front50, but it wouldn't address versions that already exist, so we might as well just fix it in the UI, since we'd need to do it there, anyway.

@jrsquared or @icfantv PTAL

### CODE REVIEW

**Comment by jrsquared** [comment]
LGTM.  Only comment is that the `jsonDiffService` now provides more than just diffing (sorting funcitons), so maybe a rename is in order to better encapsulate the total functionality, like `jsonUtilityService`

**Comment by anotherchrisberry** [comment]
Good point, thanks - I will update the name and package.

File: app/scripts/modules/core/pipeline/config/actions/history/showHistory.controller.js [hunk 1]
```diff
@@ -3,20 +3,21 @@
...
     require('core/utils/jQuery'),
-    require('./jsonDiff.service'),
+    JSON_UTILITY_SERVICE,
     require('./diffSummary.component'),
     require('./diffView.component'),
   ])
-  .controller('ShowHistoryCtrl', function ($window, pipelineConfigService, jsonDiffService,
+  .controller('ShowHistoryCtrl', function ($window, pipelineConfigService, jsonUtilityService,
                                            pipelineConfigId, currentConfig,
-                                           $uibModalInstance, $, $filter) {
```
Comment (icfantv) [inline app/scripts/modules/core/pipeline/config/actions/history/showHistory.controller.js]: how the heck did this even work before?  i thought we removed the jquery service last fall.

Comment (anotherchrisberry) [inline app/scripts/modules/core/pipeline/config/actions/history/showHistory.controller.js]: Great question - I decided not to ask and just remove it

**Pull request merged**